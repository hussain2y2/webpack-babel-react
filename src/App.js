import React from 'react'
import './assets/scss/app.scss'

class App extends React.Component{
  render() {
    return (
      <div className="app">
        <h3>Hello World!</h3>
      </div>
    )
  }
}

export default App
