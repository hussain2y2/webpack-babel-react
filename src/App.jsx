import React from 'react'
import Default from 'layouts/Default'
import '../src/assets/scss/app.scss'

class App extends React.Component{
  render() {
    return (
      <div className="app">
        <Default />
      </div>
    )
  }
}

export default App
