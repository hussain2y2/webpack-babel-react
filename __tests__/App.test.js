import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import App from 'src/App'

describe('App Component', () => {
  const wrapper = shallow(<App />)

  it('component rendered', () => {
    expect(wrapper.exists()).toBe(true)
  })

  it('claas name exist',  () => {
    expect(wrapper.find('.app').length).toBe(1)
  })

  it('renders correctly', () => {
    const snapShot = renderer.create(<App />).toJSON()
    expect(snapShot).toMatchSnapshot()
  })
})
