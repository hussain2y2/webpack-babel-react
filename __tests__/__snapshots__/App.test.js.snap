// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly 1`] = `
<div
  className="app"
>
  <div
    className="default-layout"
  >
    <nav
      className="header navbar navbar-expand-md navbar-light bg-light"
    >
      <ul
        className="mr-auto navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About Us
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/portfolio"
            onClick={[Function]}
          >
            Portfolio
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/news"
            onClick={[Function]}
          >
            News
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="row"
      style={
        Object {
          "margin": "auto",
        }
      }
    >
      <div
        className="col-12 col"
      >
        <div
          className="home"
        >
          <h4>
            Home Page
          </h4>
        </div>
      </div>
    </div>
  </div>
</div>
`;
